hs_core/hydroshare/utils.py

```diff
@@ -433,11 +433,19 @@ def copy_resource_files_and_AVUs(src_res_id, dest_res_id, set_to_private=False):
     tgt_coll = tgt_res.root_path
     for avu_name in avu_list:
         value = istorage.getAVU(src_coll, avu_name)
+        # possibly create private copy if requested
         if avu_name == 'isPublic':
             if set_to_private:
+                tgt_coll.raccess.public = False
                 istorage.setAVU(tgt_coll, avu_name, 'False')
             else:
+                tgt_coll.raccess.public = value
```

Comment:
@alvacouch Instead of directly calling ```tgt_coll.raccess.public = False``` to set public flag on ResourceAccess, I think it'll be better to call set_resource_sharing_status() since some other dependent flags such as discoverable and private may also need to be set accordingly when changing public flag. I do realize set_resource_sharing_status() is a private method and offers much more than what is needed here, so perhaps a small method on ResourceAccess model can be created to make sure all dependent flags are set accordingly as well? For example, when setting to private, make sure discoverable flag is false as well, etc.